connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -183,7 +183,7 @@ class DataFrameReader private[sql] (sparkSession: SparkSession) extends Logging
       dataSourceBuilder.setFormat(source)
       userSpecifiedSchema.foreach(schema => dataSourceBuilder.setSchema(schema.toDDL))
       extraOptions.foreach { case (k, v) =>
-        dataSourceBuilder.putOptions(k, v)
+        builder.getReadBuilder.putOptions(k, v)
```

Comment:
should modify `table` method as well?

Reply:
+1
Can you also add a simple test `reader.options().table()`?

Reply:
I found I can only add a meaningful test in server side. On client sides there is no way to verify an option has passed through.

In existing codebase, it is tested because we can do `df.queryExecution.analyzed` then get the complete plan the to fetch the options and verify it.